pkg/arvo/sys/vane/behn.hoon

```diff
@@ -45,159 +45,109 @@
-  ::  +rest: cancel the timer at :date, then adjust unix wakeup
-  ::  +wait: set a new timer at :date, then adjust unix wakeup
-  ::
-  ++  rest  |=(date=@da set-unix-wake(timers.state (unset-timer [date duct])))
-  ++  wait  |=(date=@da set-unix-wake(timers.state (set-timer [date duct])))
-  ::  +huck: give back immediately
+    [moves state]
   ::
-  ::    Useful if you want to continue working after other moves finish.
+  +|  %entry-points
   ::
-  ++  huck
-    |=  syn=sign-arvo
-    =<  [moves state]
-    event-core(moves [duct %give %heck syn]~)
-  ::  +drip:  XX
+  ++  call
+    |=  [=task error=(unit tang)]
+    ^+  this
+    ?:  ?&  ?=(^ error)
+            !?=(%wake -.task)
+        ==
+      ::  XX more and better error handling
+      ::
+      ~&  %behn-crud-not-wake^-.task
+      (emit [duct %slip %d %flog %crud -.task u.error])
+    ::
+    ?-  -.task
+      %born  this(next-wake.state ~, unix-duct.state duct)
+      %drip  (drip p.task)
+      %huck  (emit [duct %give %heck syn.task])
+      %rest  this(timers.state (unset-timer [p.task duct]))
+      %trim  this
+      %vega  this
+      %wait  this(timers.state (set-timer [p.task duct]))
+      %wake  (wake(next-wake.state ~) error)
+    ==
   ::
-  ++  drip
-    |=  mov=vase
-    =<  [moves state]
-    ^+  event-core
-    =.  moves
-      [duct %pass /drip/(scot %ud count.drips.state) %b %wait +(now)]~
-    =.  movs.drips.state
-      (~(put by movs.drips.state) count.drips.state mov)
-    =.  count.drips.state  +(count.drips.state)
-    event-core
   ::  +take-drip:  XX
   ::
   ++  take-drip
     |=  [num=@ud error=(unit tang)]
-    =<  [moves state]
-    ^+  event-core
+    ^+  this
     =/  drip  (~(got by movs.drips.state) num)
-    =.  movs.drips.state  (~(del by movs.drips.state) num)
-    =/  =move
-      =/  card  [%give %meta drip]
-      ?~  error
-        [duct card]
-      =/  =tang
-        (weld u.error `tang`[leaf/"drip failed" ~])
-      ::  XX should be
-      ::  [duct %hurl fail/tang card]
-      ::
-      [duct %pass /drip-slog %d %flog %crud %drip-fail tang]
-    event-core(moves [move moves])
-  ::  +trim: in response to memory pressue
+    %-  emit(movs.drips.state (~(del by movs.drips.state) num))
+    =/  card  [%give %meta drip]
+    ?~  error
+      [duct card]
+    =/  =tang
+      (weld u.error `tang`[leaf/"drip failed" ~])
+    ::  XX should be
+    ::  [duct %hurl fail/tang card]
+    ::
+    [duct %pass /drip-slog %d %flog %crud %drip-fail tang]
+  ::
+  +|  %tasks
   ::
-  ++  trim  [moves state]
-  ::  +vega: learn of a kernel upgrade
+  ::  +drip:  XX
+  ::
+  ++  drip
+    |=  vax=vase
+    ^+  this
+    %.  [duct %pass /drip/(scot %ud count.drips.state) %b %wait +(now)]
+    %=  emit
+      movs.drips.state   (~(put by movs.drips.state) count.drips.state vax)
+      count.drips.state  +(count.drips.state)
+    ==
   ::
-  ++  vega  [moves state]
   ::  +wake: unix says wake up; process the elapsed timer and set :next-wake
```

Comment:
this looks like it no longer sets .next-wake, since that's now done in +abet

Reply:
good catch.